public/script.js

```diff
@@ -1725,39 +1701,37 @@ async function Generate(type, automatic_trigger, force_name2) {
             generatedPromtCache += cycleGenerationPromt;
             if (generatedPromtCache.length == 0) {
                 if (main_api === 'openai') {
-                    generateOpenAIPromptCache(charPersonality, topAnchorDepth, anchorTop, anchorBottom);
+                    generateOpenAIPromptCache(charPersonality, topAnchorDepth, anchorTop, bottomAnchorThreshold, anchorBottom);
                 }
 
                 console.log('generating prompt');
                 chatString = "";
                 arrMes = arrMes.reverse();
-                let is_add_personality = false;
                 arrMes.forEach(function (item, i, arr) {//For added anchors and others
 
-                    if (i >= arrMes.length - 1 && $.trim(item).substr(0, (name1 + ":").length) != name1 + ":") {
+                    if (i === arrMes.length - 1 && $.trim(item).substr(0, (name1 + ":").length) != name1 + ":") {
                         if (textareaText == "") {
                             item = item.substr(0, item.length - 1);
                         }
                     }
-                    if (i === arrMes.length - topAnchorDepth && count_view_mes >= topAnchorDepth && !is_add_personality) {
-                        is_add_personality = true;
```

Comment:
`i >= arrMes.length - 1` happens only once per loop (now `i === arrMes.length - 1`, making it more obvious), so there is no need to remember if personality was added in a previous iteration.